akka-actor/src/main/scala/akka/io/dns/protocol/DnsRecordType.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.io.dns.protocol
+
+import akka.annotation.InternalApi
+import akka.util.{ ByteIterator, ByteStringBuilder, OptionVal }
+
+/**
+ * DNS Record Type
+ */
+final case class DnsRecordType(code: Short, name: String)
+
+object DnsRecordType {
+
+  // array for fast lookups by id
+  // wasteful, but we get trivial indexing into it for lookup
+  private final val lookupTable = Array.ofDim[DnsRecordType](256)
```

Comment:
this allows us lookups by directly indexing into the array, so the `apply()` should be very efficient